utils_linux.go

```diff
@@ -419,7 +419,7 @@ func startContainer(context *cli.Context, spec *specs.Spec, action CtAct, criuOp
 	}
 
 	if notifySocket != nil {
-		err := notifySocket.setupSocket()
+		err := notifySocket.setupSocket(spec.Process.User)
```

Comment:
How does this handle user namespaces? I have a feeling you'd need to use the host-uid/gid here rather than the container one.

Reply:
We could probably just change the mode to 0777, which is the same mode used on the host by systemd for `/run/systemd/notify`.  @jburianek what do you think?

Reply:
@cyphar Good point, I forgot about user namespaces.

@giuseppe That seems reasonable, I've modified the PR with your suggestion.